### ISSUE TITLE
[question/request] include port requirements in Flatpak

### ISSUE THREAD

**Issue by omichalek** [flatpak/flatpak#1759]
I recently discovered that my Flatpaked Steam client was being blocked by firewall, which led me to the question: _**should Flatpak applications not document the ports they need in order to operate?**_ Not necessarily for flatpak to bind the traffic to them as [#1688](https://github.com/flatpak/flatpak/issues/1688) describes (even if that would make sense too) but at least the system would have a way to automatically open up those ports in firewall.
Even without system integration like this the user would benefit greatly by having the port requirements for all the applications available in one place!
.___
edited for clarity, added this:
The main issue that I can imagine (having user-only knowledge of flatpak): some applications will need to change the ports after being installed. While you can always define a greater range of ports in the metadata, it is probably better to either
- store this information somewhere where the application can change it by itself (not in the read-only metadata)
- or the application should be able to prompt user with `flatpak override` request or similar to change its own port configuration

**Comment by TingPing** [comment]
Why would Flatpak know that information, it isn't useful information to `flatpak` itself so at that point it just becomes a documentation issue for apps themselves.

**Comment by omichalek** [comment]
@TingPing making (network dependent) applications work behind firewall does sound to me like an issue for the tool that you use to install your applications.
- Even if opening up the firewall was not a task for the flatpak command itself, it still makes a lot of sense to require that information from the applications in a pre-defined way, because having a standard place to store the port numbers and protocols would allow others to build on top of that and eg. make application-aware firewalls where user can simply enable (or disable) applications that are relevant to them.
- Also, if this was just a 'documentation issue' for the applications it is completely right to do that I would say: in the same sense that they should use standard places to store the configuration data, why would you not require them to document what ports they need?

**Comment by alexlarsson** [comment]
I don't think this is really the domain of flatpak, because we currently do nothing wrt network ports or any fine-grained network details. As such, there is zero gain or testing of what applications would list for such information, and it is unlikely to be filled in or be correct.
